app/controllers/supervisor/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Supervisor::CommentsController < Supervisor::BaseController
+
+  def create
+    comment = Comment.new(comment_params)
+
+    if comment.save
```

Comment:
I like this. Also, I doubt that a comment without text should be saved in the database. I know I do save it right now, but that is probably not as it is meant to be. 
To be continued.